### DISCUSSION TITLE
Manually Create Menu Shape instance

### DISCUSSION THREAD

**Discussion by effapp** [OrchardCMS/OrchardCore#16587] [Q&A]
Hi,
    How to manually create Menu Shape instace variable "menu" in follow code:

<nav class="{{ cssClasses }}">
    {% shape "menu", alias: menuAlias %}
</nav>

**Comment by Piedone** [comment]
Are you looking for [`shape_render`](https://docs.orchardcore.net/en/latest/reference/modules/Liquid/#shape_render)?

**Reply by effapp** [reply]
No.
I am looking for example of code to create menu shape instance that is input of  "shape" of liquid.

**Reply by effapp** [reply]
Hi @Piedone ,
I want to create a object of Menu shape  from taxonomy object by coding, then using {% shape "menu", alias: menuAlias %} to render menu in the widget. But I can't find code example of creating object of Menu shape
Could you provide a pieces of codes example of creating object of Menu shape by "_shapeFactory.CreateAsync", please?

**Reply by Piedone** [reply]
The `shape` tag works as you use it, and doesn't support having a shape object as the input. That's what `shape_render` is for. Furthermore, [`shape_new`](https://docs.orchardcore.net/en/latest/reference/modules/Liquid/#shape_new) can create a shape object for it, which is equivalent to `_shapeFactory.CreateAsync`.

I recommend you to check out these sections: https://docs.orchardcore.net/en/latest/reference/modules/Liquid/#shape-filters and https://docs.orchardcore.net/en/latest/reference/modules/Liquid/#shape-tags.
